hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/AttributeSearchService.kt

```diff
@@ -45,13 +48,36 @@ class AttributeSearchService(
     }.search(pageable)
 
   private fun BoolQueryBuilder.search(pageable: Pageable): Page<Prisoner> {
-    val searchSourceBuilder = SearchSourceBuilder().query(this)
+    val searchSourceBuilder = pageable.searchSourceBuilder(this)
     val searchRequest = SearchRequest(arrayOf(PRISONER_INDEX), searchSourceBuilder)
     val searchResponse = elasticsearchClient.search(searchRequest)
     val results = searchResponse.hits.hits.asList().map { mapper.readValue(it.sourceAsString, Prisoner::class.java) }
     return PageImpl(results, pageable, searchResponse.hits.totalHits?.value ?: 0)
   }
 
+  private fun Pageable.searchSourceBuilder(queryBuilder: BoolQueryBuilder): SearchSourceBuilder {
+    val sortBuilders = sort.map {
+      FieldSortBuilder(getSearchableProperty(it)).order(SortOrder.fromString(it.direction.name))
+    }.toList()
+    return SearchSourceBuilder().apply {
+      query(queryBuilder)
+      size(pageSize)
+      from(offset.toInt())
+      sortBuilders.takeIf { it.isNotEmpty() }
+        ?.forEach { sort(it) }
+        ?: sort("prisonerNumber")
+    }
+  }
+
+  private fun getSearchableProperty(it: Sort.Order): String? {
+    val type = attributes[it.property] ?: throw AttributeSearchException("Sort attribute '${it.property}' not found")
+    return when {
+      it.property == "prisonerNumber" -> "prisonerNumber"
+      type.simpleName == "String" -> "${it.property}.keyword"
+      else -> it.property
+    }
+  }
```

Comment:
Strings need sorting by the keyword - except for prisonerNumber (which I guess is because it's the `@Id` column).